src/corehost/cli/test/nativehost/comhost_test.cpp

```diff
@@ -109,22 +112,16 @@ bool comhost_test::concurrent(const pal::string_t &comhost_path, const pal::stri
     for (int i = 0; i < count; ++i)
         activations.push_back(std::async(std::launch::async, activate_class, comhost, clsid));
 
+    std::stringstream ss;
     bool succeeded = true;
     for (int i = 0; i < count; ++i)
     {
         HRESULT hr = activations[i].get();
+        log_activation(clsidVect.data(), i + 1, count, hr, ss);
```

Comment:
Doesn't this suffer from the same issues as before except not the the writing could be interleaved in the `stringstream`?

Reply:
The interleaving was between the activated COM class (does a `Console.WriteLine` on instantiation) and the logging here. This change makes it so that the test host stores its messages in the `stringstream` and only prints to stdout at the end (after waiting for all the activations so they would all have printed their output already)